src/main/java/org/broadinstitute/hellbender/tools/exome/ExonDB.java

```diff
@@ -0,0 +1,299 @@
+     *
+     * <p>The returned collection
+     * might not be modifiable and might change if this {@link ExonDB} is mutable.
+     *
+     * @param overlapRegion the query location.
+     * @return never {@code null} but an empty list if there is no such exons.
+     * @throws IllegalArgumentException if {@code overlapRegion} is {@code null}.
+     */
+    default List<E> exons(final GenomeLoc overlapRegion) {
+        final IndexRange range = indexRange(overlapRegion);
+        return exons().subList(range.from,range.to);
+    }
+
+    /**
+     * Returns exons within an index interval.
+     *
+     * @param range the target index range.
+     * @return never {@code null} but perhaps an empty list.
+     * @throws IllegalArgumentException if {@code range} is {@code null} or
+     *   {@code range} contains invalid indices in this exon collection.
+     */
+    default List<E> exons(final IndexRange range) {
+        if (range == null) {
+            throw new IllegalArgumentException("range cannot be null");
+        } else if (!range.isValidLength(exonCount())) {
+            throw new IllegalArgumentException(
+                    String.format("index range '%s' is invalid for an exonCount of %d.", range, exonCount()));
+        }
+        return exons().subList(range.from,range.to);
+    }
+
+
+    /**
+     * Returns exons within an index interval.
+     *
+     * @param from the index of the first requested exon.
+     * @param to   the index of the exon right after the last one requested.
+     * @return never {@code null} but perhaps an empty list.
+     * @throws IllegalArgumentException if {@code from} and {@code to} is not a valid exon index range.
+     */
+    default List<E> exons(final int from, final int to) {
+        if (from < 0) {
+            throw new IllegalArgumentException(
+                    String.format("from index (%d) cannot be negative",from));
+        } else if (to > exonCount()) {
+            throw new IllegalArgumentException(
+                    String.format("to index (%d) cannot larger than the exon-count (%d)",to,exonCount()));
+        }
+        return exons().subList(from,to);
+    }
+
+    /**
+     * Returns index range of exons that overlap a location.
+     * <p>
+     * If no exon overlap the location, the returned range would have size 0 and would indicate
+     * the insert index where such exons would be found if they were part of the collection.
+     * </p>
+     *
+     * @param location the target location.
+     * @throws IllegalArgumentException if {@code location} is {@code null}.
+     * @return never {@code null} but perhaps an empty range.
+     */
+    IndexRange indexRange(final GenomeLoc location);
+
+    /**
+     * Returns a list of all exons sorted by genomic coordinates.
+     *
+     * @return never {@code null} but perhaps empty if there is no exon in this collection.
+     */
+    List<E> exons();
```

Comment:
document whether this method returns the live list from the DB or a copy or an unmodifiable view. The API client needs to know this.